### ISSUE TITLE
Multiple questions

### ISSUE THREAD

**Issue by akation** [rainlab/user-plugin#475]
Hello,

I have some questions, I hope you guys could help me out.

1. In reset password form (I have a custom one, how can I handdle form success? Im handdling form fail messages with ajaxErrorMessage but I want to handle the message when the form is sent.
2. In register (I'm using a custom form), how can I make name and surname required? 
3. I would like to have a custom form for password reset (As I did with register and login) any example you could show me?
4. I have a profile form where the user can update its data, I have custom fields like Address 1, Address 2, Zip code. How can i add those custom fields to the user data? Also can you show me an example of something similar? 

Sorry for all the questions, thank you beforehand.

**Comment by bennothommo** [comment]
@akation

1. Do you wish to customise the success message content, or are you attempting to do something else?
2. You have 2 options - you can modify the validation rules of the User model through [extending the User model](https://octobercms.com/docs/plugin/extending#extending-user-model) or you could simply add a `required` attribute to the fields in your custom form.
3. Unfortunately I cannot think of any examples off the top of my head, but it's the same idea as customising the register and login forms as you have done already.
4. You could take a look at the [User Plus plugin](https://github.com/rainlab/userplus-plugin) which shows an example of how to add additional custom fields to the user data.

Hope that helps :).

**Comment by akation** [comment]
Hello,

Thank you for the quirck answer!

1. I just want to show the success message, I know how to show the fail message, but dont know how to show the opposite with ajax.
3. With register and login I did not developed the logic, I just set the same names and attributes I saw in the user components, As there is not component for set new password, I dont know if theres a method i could use. 

Thank you so much for the help!
